src/core/seat/seat.cpp

```diff
@@ -289,6 +289,26 @@ uint32_t wf::seat_t::get_modifiers()
     return current_keyboard ? current_keyboard->get_modifiers() : 0;
 }
 
+std::string wf::seat_t::convert_keycode(uint32_t keycode)
```

Comment:
This doesn't need to be in core because plugins can access the `wlr_seat` from core (get_current_seat), and then with wlr_seat_get_keyboard() you can find the currently active keyboard.

I wonder why you say it doesn't work in all cases though? Does it work with for ex. umlaut characters like äüö with the compose key?

Reply:
I see, this makes sense.

Now I realized that compose / dead keys are not working on my Wayfire session at all, but working in a nested session, so this might be a problem with my layout settings.